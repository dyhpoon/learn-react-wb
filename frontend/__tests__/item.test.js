import ItemComponent from '../components/Item';
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

const fakeItem = {
  id: 'abc123',
  title: 'the title',
  price: 3013,
  description: 'this is a description',
  image: 'dog.jpg',
  largeImage: 'largedog.jpg',
}

describe('<Item/>', () => {
  it('renders and displays properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem}/>)
    expect(wrapper.find('PriceTag').children().text()).toBe('$30.13')
    expect(wrapper.find('Title a').text()).toBe(fakeItem.title)
    expect(wrapper.find('img').props().src).toBe(fakeItem.image)
    expect(wrapper.find('img').props().alt).toBe(fakeItem.title)
  })

  it('renders out the buttons properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem}/>)
    const buttonList = wrapper.find('.buttonList');
    expect(buttonList.children()).toHaveLength(3)
    expect(buttonList.find('Link').exists()).toBeTruthy()
  })

  it('snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})