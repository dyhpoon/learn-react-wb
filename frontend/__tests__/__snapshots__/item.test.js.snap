// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="the title"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "abc123",
          },
        }
      }
    >
      <a>
        the title
      </a>
    </Link>
  </Title>
  <PriceTag>
    $30.13
  </PriceTag>
  <p>
    this is a description
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "abc123",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="abc123"
    />
    <DeleteItem
      id="abc123"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
